src/main/java/ca/bc/gov/backendstartapi/repository/UserRepository.java

```diff
@@ -23,13 +24,13 @@ public class UserRepository {
    * @param userDto user to be saved
    * @return the new saved user.
    */
-  public Mono<UserDto> save(UserDto userDto) {
+  public UserDto save(UserDto userDto) {
     UserDto saved = users.put(userDto.hashCode(), userDto);
     if (!Objects.isNull(saved)) {
-      return Mono.error(new UserExistsException());
+      throw new UserExistsException();
```

Comment:
Much better. Thanks.

Reply:
The documentations is quite confusing. First, it seems that unchecked exceptions should not be added with @throws tag, but a few lines bellow, it seems it can be added as **merely a device meant to indicate this exception is part of the contract**. Anyways, I think is a good idea to indicate unchecked exceptions as well.